### PR TITLE
More USB code cleanup

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: libs/usb/HotplugAgent.cpp [hunk 1]
```diff
@@ -77,25 +77,23 @@ AsyncronousLibUsbAdaptor *HotplugAgent::GetUSBAdaptor() const {
...
 
   m_use_hotplug = HotplugSupported();
   OLA_DEBUG << "HotplugSupported(): " << m_use_hotplug;
-  if (m_use_hotplug) {
 #ifdef HAVE_LIBUSB_HOTPLUG_API
+  if (m_use_hotplug) {
     m_usb_thread.reset(new ola::usb::LibUsbHotplugThread(
           m_context, hotplug_callback, this));
-#else
-    OLA_FATAL << "Mismatch between m_use_hotplug and HAVE_LIBUSB_HOTPLUG_API";
```
Comment (peternewman) [inline libs/usb/HotplugAgent.cpp]: Don't we want the error message still?

File: libs/usb/Types.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+bool USBDeviceID::operator<(const USBDeviceID &id) const {
+  if (bus_number < id.bus_number) {
+    return true;
+  } else if (bus_number == id.bus_number) {
+    return device_address < id.device_address;
+  }
+  return false;
+}
+
+ostream& operator<<(ostream& os, const USBDeviceID &id) {
```
Comment (peternewman) [inline libs/usb/Types.cpp]: Is this not similar duplication to the other << operator you added just recently?

**Comment by peternewman** [comment]
LGTM

File: libs/usb/HotplugAgent.cpp (hunk 1 above)
Comment (nomis52) [inline libs/usb/HotplugAgent.cpp]: There shouldn't be any way for this message to occur so I removed it to make the code simpler.

File: libs/usb/Types.cpp (hunk 2 above)
Comment (nomis52) [inline libs/usb/Types.cpp]: Which one? I can't find one.

File: libs/usb/HotplugAgent.cpp (hunk 1 above)
Comment (peternewman) [inline libs/usb/HotplugAgent.cpp]: Okay, I thought it may be worth it for the edge case.

File: libs/usb/Types.cpp (hunk 2 above)
Comment (peternewman) [inline libs/usb/Types.cpp]: I found it eventually: https://github.com/OpenLightingProject/ola/pull/926/files#diff-034547a73202f2f4f37ca79aabba4010R39

Comment (peternewman) [inline libs/usb/Types.cpp]: Ah, it's the same one, it's just no longer in the header...

**Pull request merged**